enricher/api/src/main/java/io/fabric8/maven/enricher/api/Enricher.java

```diff
@@ -63,14 +37,12 @@
      *
      * @param builder the build to examine and add to
      */
-    void addMissingResources(KubernetesListBuilder builder);
+    void enrich(PlatformMode platformMode, KubernetesListBuilder builder);
```

Comment:
Looks very nice now ! It has now the two inner passes from the initial suggestion.

Wdyt about renaming these two methods to "create" (was "enrich") and to "enrich" (was "adapt") as suggested in #678 ?

That way we can clearly indicate the intension of this API, too.